### DISCUSSION TITLE
Whitelist URL with a dynamic GET parameter

### DISCUSSION THREAD

**Discussion by millerbean16** [etsy/open-api#837] [Q&A]
I have tried to add the following URLs to the whitelist
https://localhost:44352/api/plugin/EtsyConnectCallback?userId=*
https://localhost:44352/api/plugin/EtsyConnectCallback?userId=

But when using the redirect it needs to be dependent on a dynamic userID, which will change on each user connecting from our shop to Etsy

eg.
https://localhost:44352/api/plugin/EtsyConnectCallback?userId=111
https://localhost:44352/api/plugin/EtsyConnectCallback?userId=222

But when calling with these redirect urls and with the whitelisted urls above we still get the "The requested redirect URL is not permitted" when trying to grant access to Etsy from our shop.

How can we add a whitelist url, so it is not dependent on a specific userID ?

Do you in any way support wildcards in the url and how ?

If not, how is it possible to see which user it is, when you are redirecting the user back to our shop, if we are not allowed to have a dynamic GET parameter (userid) in the url ?

Any suggestions would be much appreciated 👌

**Comment by turboturtleD** [comment]
I think the prefix on the token is the user_id for the shop.

**Comment by Gareth064** [comment]
When you make an authorization request, you have to provide a State parameter. Make that state code unique to each request (per user). The response you get back from Etsy will include that same State parameter which you can use to match back to the user request you sent

**Reply by millerbean16** [reply]
Thanks - that was the answer I was looking for!
